test/Polly.Extensions.Tests/Telemetry/TelemetryListenerImplTests.cs

```diff
@@ -453,9 +453,33 @@ public void PipelineExecuted_ShouldBeSkipped()
         events.Should().HaveCount(0);
     }
 
+    [Fact]
+    public void SeverityProvider_EnsureRespected()
```

Comment:
Is it worth adding a test that explicitly validates the severity isn't changed if `SeverityProvider` is null?

Reply:
Done.